### PR TITLE
Feature/zenko 1402 move data backends

### PR DESCRIPTION
**Pull request by dora-korpar**
Beginning the work of moving the data backends to Arsenal.
This PR includes the external backend clients, the necessary supporting functions, and the existing unit tests for both of those.
Most files have simply been copied over (with adjusted requirements). The only code changes beyond requirements have been to `getVault.js`, which was copied over from `vault.js` (https://github.com/scality/cloudserver/blob/development/8.1/lib/auth/vault.js) in CloudServer. In order to circumvent the problem of assuming the use of the CloudServer config, the code was wrapped in a function.
The other change is that a config parameter was introduced to several client function, as they too require the CloudServer config.

### CODE REVIEW

**Comment by dora-korpar** [comment]
@bert-e reset

**Review by JianqinWang — APPROVED** [review]

**Review by alexanderchan-scality — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -36,6 +39,7 @@
     "socket.io-client": "~2.2.0",
     "utf8": "3.0.0",
     "uuid": "^3.0.1",
+    "vaultclient": "scality/vaultclient#754b6e1",
```
Comment (vrancurel) [inline package.json]: We introduce here a circular dependency: only cloudserver shall be able to have vaultclient in its package.json. The vaultclient lib instance shall be passed from cloudserver to Arsenal (just like sproxydclient and bucketclient)

**Review by vrancurel — DISMISSED** [review]

**Comment by vrancurel** [comment]
where are we now ?

**Comment by dora-korpar** [comment]
The vaultclient dependency was removed, and this is ready to go, there are just a few tests failing in CloudServer that I need to verify before merging this

**Review by rahulreddy — APPROVED** [review]

**Comment by dora-korpar** [comment]
@bert-e approve

**Pull request merged**